.github/workflows/knative-downstream.yaml

```diff
@@ -80,19 +80,34 @@ jobs:
       uses: actions/setup-go@v2
       with:
         go-version: 1.16.x
+
     - name: Install Dependencies
       run: |
         go get github.com/google/go-licenses
-    - name: Checkout Upstream
+
+    - name: Checkout upstream at ${{ github.base_ref }}
       uses: actions/checkout@v2
       with:
+        fetch-depth: 0
         path: ./src/knative.dev/${{ github.event.repository.name }}
-    - name: Checkout Downstream
+
+    - name: Merge upstream
+      if: github.event_name == 'pull_request'
+      shell: bash
+      working-directory: ./src/knative.dev/${{ github.event.repository.name }}
+      run: |
+         git remote add upstream https://github.com/${{ github.repository }}.git
+         git fetch upstream ${{ github.base_ref }}
+         git pull --no-rebase --no-commit upstream ${{ github.base_ref }}
```

Comment:
I think what you are going to be doing is checking out the fork, and then pulling the forks branch that the pr is based on and what you are trying to do is checkout the original repo and try to merge the fork onto it? I am not sure this is needed, if the repo has broken the API for the downstream, it should have been reported, and there would be a conflict?

Reply:
This pulls in the latest commits from `HEAD` into the PRs changes. We already do this prior to running unit tests etc.